src/scripts/main.js

```diff
@@ -1,3 +1,266 @@
 'use strict';
 
-// write your code here
+const gameContainer = document.querySelector('.container');
+const startButton = gameContainer.querySelector('.button');
+const cells = gameContainer.querySelectorAll('.field-cell');
+const messageStart = gameContainer.querySelector('.message-start');
+const messageWin = gameContainer.querySelector('.message-win');
+const messageLose = gameContainer.querySelector('.message-lose');
+const gameScore = gameContainer.querySelector('.game-score');
+const gameRow = gameContainer.querySelectorAll('tr');
+const probabilityGeneratingNumber = 0.1;
+const numberFour = 4;
+const numberTwo = 2;
+const moveLeft = 'ArrowLeft';
+const moveRight = 'ArrowRight';
+const moveUp = 'ArrowUp';
+const moveDown = 'ArrowDown';
+const winnerNumber = '2048';
+let movePossible = false;
+let adjacentCells = false;
+
+function generateRandom() {
+  const randomI = Math.floor(Math.random() * cells.length);
```

Comment:
Recursive function 'generateRandom' can lead to maximum call stack size exceeded error if all cells are filled. I suggest you to first check if there is an available space before calling the function and then generate a random number for an empty cell.